agent.go

```diff
@@ -246,10 +247,25 @@ func (c *controller) agentInit(bindAddrOrInterface, advertiseAddr string) error
 	}
 
 	keys, tags := c.getKeys(subsysGossip)
+
+	// Make the hostname unique in the gossip cluster by appending a 12
+	// byte random ID. Between distros there is no consensus if the
+	// hostname should be an FQDN or shortname. Limit the hostname to
+	// RFC 1123 compliant 63 bytes
 	hostname, _ := os.Hostname()
+	id := stringid.TruncateID(stringid.GenerateRandomID())
+
+	var nodeName string
+	if len(hostname)+len(id) > 63 {
```

Comment:
But hostnames are more meaningful and if somebody used hostnames in their cluster with same prefix then there is a potential that we truncate the differentiating parts and keep the same part and it would be hard to read logs if that is the case. Do we really need to keep node name length consistent across the cluster?